core/src/main/java/org/elasticsearch/action/admin/cluster/settings/SettingsUpdater.java

```diff
@@ -49,15 +49,12 @@ synchronized Settings getPersistentUpdate() {
     }
 
     synchronized ClusterState updateSettings(final ClusterState currentState, Settings transientToApply, Settings persistentToApply) {
-        boolean changed = false;
         Settings.Builder transientSettings = Settings.builder();
         transientSettings.put(currentState.metaData().transientSettings());
-        changed |= clusterSettings.updateDynamicSettings(transientToApply, transientSettings, transientUpdates, "transient");
-
         Settings.Builder persistentSettings = Settings.builder();
         persistentSettings.put(currentState.metaData().persistentSettings());
-        changed |= clusterSettings.updateDynamicSettings(persistentToApply, persistentSettings, persistentUpdates, "persistent");
-
+        boolean changed = clusterSettings.updateDynamicSettings(transientToApply, transientSettings, transientUpdates, "transient") |
```

Comment:
This isn't a hot code path so I'm quite willing to try and write the most readable code possible and let the optimizer try and optimize out the extra OR operation if it is capable of it.